helm/provider_test.go

```diff
@@ -44,36 +44,58 @@ func TestMain(m *testing.M) {
 		panic("Could not create temporary directory for helm config files")
 	}
 
-	os.Setenv("HELM_REPOSITORY_CONFIG", filepath.Join(home, "config/repositories.yaml"))
-	os.Setenv("HELM_REPOSITORY_CACHE", filepath.Join(home, "cache/helm/repository"))
-	os.Setenv("HELM_REGISTRY_CONFIG", filepath.Join(home, "config/registry.json"))
-	os.Setenv("HELM_PLUGINS", filepath.Join(home, "plugins"))
-	os.Setenv("XDG_CACHE_HOME", filepath.Join(home, "cache"))
+	err = os.Setenv("HELM_REPOSITORY_CONFIG", filepath.Join(home, "config/repositories.yaml"))
+	if err != nil {
```

Comment:
Thanks for adding this error handling! 

Complete tangent, but I went digging in the Go source code to see under what condition `os.Setenv` would _actually_ fail. `os.Setenv` turns out just to be a proxy to `syscall.Setenv` which (on *nix) will return `syscall.EINVAL` if the key contains any null characters or `=`, or the value contains any null character _unless_ the [platform is Plan 9](https://golang.org/src/syscall/env_unix.go#L106) 😄. tl;dr we're pretty safe errors here.